### DISCUSSION TITLE
Consulta en herencia de excepciones - Monedero

### DISCUSSION THREAD

**Discussion by mbeorlegui** [dds-jv/foro#366] [Consultas de objetos] [answered]
Buenas, estoy teniendo una duda sobre qué tan bien o mal está heredar excepciones.

Por ejemplo, en nuestro caso `SaldoMenorException` (y varias otras) hereda de `RuntimeException`, y eso hace que en la primer clase sólo te quede el constructor con un super, cosa que parece ser un Code Smell.

Por otro lado, pareciera que en este caso es para particularizar una/s excepcion/es, cosa que me parece bastante más expresivo, y por ende válido de hacer.

Quería saber cómo debemos accionar en este caso.

Saludos y buen finde!

**Comment by gastonprieto** [comment, marked as answer]
Buenas @mbeorlegui !

Claro, este es uno de los casos particulares, en Java es el único mecanismo para realizar excepciones especificas, así que es correcto realizar una subclasificación para definir tus propias excepciones, incluso es mejor que usar RuntimeException.
